cmd/kube-apiserver/app/server.go

```diff
@@ -194,6 +194,19 @@ func CreateServerChain(runOptions *options.ServerRunOptions, stopCh <-chan struc
 	return aggregatorServer.GenericAPIServer, nil
 }
 
+func CreateListener(addr string) (net.Listener, error) {
+	if len(addr) == 0 {
+		return nil, fmt.Errorf("address cannot be empty")
+	}
+
+	ln, err := net.Listen("tcp", addr)
```

Comment:
won't this fail on an empty address as well?

Reply:
I look into net.Listen. And find it fail with an empty addr. So I will remove the check upper.